### PR TITLE
Signal Broker distribution support

### PR DESCRIPTION
**Pull request by oltarasenko**
1. Use erlang-node-discovery to discover nodes in the cluster
2. Read interfaces.json in order to get nodes for connection
3. Use vm.args to set node names and cookies (from ENV)
4. Adapt tests (master_node is now a required setting)

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -24,6 +24,8 @@ RUN rm -rf deps
 RUN rm -rf apps/app_ngcan/priv
 RUN mix deps.get
 # RUN mix test --only success
+
+# ACHTUNG: This builds a release with dev options
```
Comment (AleksandarFilipov) [inline docker/Dockerfile]: Ops this isn't on purpose....

File: apps/app_unixds/test/app_unixds_test.exs [hunk 2]
```diff
@@ -223,6 +223,16 @@ defmodule UnixDSTest do
...
+    setup do
+      Application.ensure_all_started(:erlang_node_discovery)
+
+      on_exit(fn ->
+        Application.stop(:erlang_node_discovery)
+      end)
+
+      :ok
+    end
+
```
Comment (AleksandarFilipov) [inline apps/app_unixds/test/app_unixds_test.exs]: Not clear to me why this is needed here.

File: apps/grpc_service/lib/grpc_service/application.ex [hunk 3]
```diff
@@ -38,6 +55,7 @@ defmodule GRPCService.Application do
     Supervisor.start_link(children, opts)
   end
 
+  # This one returns atom not a pid.. (Probably process name).
```
Comment (enmasse) [inline apps/grpc_service/lib/grpc_service/application.ex]: Nit: Add @spec?

**Comment by oltarasenko** [comment]
Closed in favor of https://github.com/volvo-cars/signalbroker-server/pull/51

**Pull request closed without merging**